### ISSUE TITLE
💅 Imported npm modules should not require .js extensions

### ISSUE THREAD

**Issue by jeremytenjo** [biomejs/biome#8245]
### Environment information

```bash
CLI:
  Version:                      2.3.7
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.10.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.6.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  false
  Enabled rules:
    complexity/noBannedTypes
    complexity/noUselessCatch
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    correctness/noChildrenProp
    correctness/noConstAssign
    correctness/noInvalidConstructorSuper
    correctness/noPrecisionLoss
    correctness/noUndeclaredVariables
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnusedImports
    correctness/noUnusedVariables
    correctness/useHookAtTopLevel
    correctness/useImportExtensions
    correctness/useJsxKeyInIterable
    correctness/useValidTypeof
    security/noBlankTarget
    security/noDangerouslySetInnerHtmlWithChildren
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useConst
    suspicious/noCommentText
    suspicious/noDebugger
    suspicious/noDuplicateJsxProps
    suspicious/noExtraNonNullAssertion
    suspicious/noMisleadingInstantiator
    suspicious/noNonNullAssertedOptionalChain
    suspicious/noUnsafeDeclarationMerging
    suspicious/noVar

Workspace:
  Open Documents:
```

### Rule name

useImportExtensions

### Playground link

https://biomejs.dev/playground/

### Expected result

When importing packages from npm (e.g., @useweb/ui), Storybook/Vite/Next.js throws errors because .js extensions are automatically appended to import paths. Many npm packages do not ship files with .js extensions at the resolved build path, causing imports such as:

`import ActionBox from '@useweb/ui/ActionBox'`

to fail with:

`import ActionBox from '@useweb/ui/ActionBox.js'`

Expected Behavior
Imports from npm packages should resolve without forcing a .js extension, using the module’s own exports and package.json exports field.

`biome.json`

 `"useImportExtensions": {
          "level": "error",
          "options": {
            "forceJsExtensions": true
          }
        },`

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ruidosujeira** [comment]
From what I see, with forceJsExtensions: true the rule is currently applying .js fixes to bare specifiers from npm package, even when the real file does not exist...

**Comment by jeremytenjo** [comment]
It seems to work with a clean repo 

<img width="762" height="283" alt="Image" src="https://github.com/user-attachments/assets/ac219c8a-aceb-4ad7-bbc1-972f08883c16" />

Could be something with my ts config then. 

```json{
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "lib": ["DOM", "DOM.Iterable", "ESNext"],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "ESNext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "noImplicitAny": false,
    "jsx": "react-jsx",
    "types": ["node"],
    "paths": {
      "@/*": ["./*"],
      "*": ["./devtools/ts/declaredModules/*"]
    },
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ]
  },
  "include": [
    "data",
    "lib",
    "app",
    "devtools",
    "devtools/ts/declaredModules/*",
    "marketing",
    ".next/types/**/*.ts",
    ".next/dev/types/**/*.ts",
    "devtools/prettier/prettier.config.js"
  ],
  "exclude": ["node_modules", ".next"]
}

**Comment by dyc3** [comment]
@jeremytenjo Please provide a playground link, or a reproduction repo. Instructions here: https://github.com/biomejs/biome/issues/8245#issuecomment-3572238595

**Comment by jeremytenjo** [comment]
I could not repro with a clean repo so I made my project public for now

https://github.com/jeremytenjo/official-pet-portraits.

1. git clone 
2. run npm i 
3. open NewsletterPreview.tsx 

You should see the false positives

<img width="912" height="179" alt="Image" src="https://github.com/user-attachments/assets/d5a0d4ff-6e19-4f99-8be7-46a1cf305b2f" />

Thank you! @dyc3

**Comment by dyc3** [comment]
You are using the `forceJsExtensions` option. If you disable that, it will go away.

If you are looking for a way to enforce `.ts` imports are `.js` imports, then see #7734. The feature for that will be in the next minor.

**Comment by jeremytenjo** [comment]
But why don't I get the error in a clean repo?

<img width="1617" height="490" alt="Image" src="https://github.com/user-attachments/assets/d35bd533-807e-4411-ab55-1c3c1b85eb26" />

https://github.com/jeremytenjo/biome-repro-1764010749511/blob/main/src/index.ts

**Comment by ematipico** [comment]
Because they aren't the same. You also have a TypeScript configuration that wasn't added to the reproduction

**Comment by ematipico** [comment]
Because they aren't the same. You also have a TypeScript configuration that wasn't added to the reproduction

**Comment by arendjr** [comment]
It seems this isn't really a bug then, but rather a problem with the configuration, given that it doesn't happen with a clean repository. Indeed the `forceJsExtensions` option should be double-checked. If you're still having issues, feel free to open a Help topic on Discord.

Re @ruidosujeira:

> From what I see, with forceJsExtensions: true the rule is currently applying .js fixes to bare specifiers from npm package, **even when the real file does not exist...**

That's intentional. The problem with the decision from the TypeScript team is that when you're importing `.js` files from TypeScript files, you never know if the files exist...
